### PR TITLE
[18.09] Move the tar creation to the image directory

### PR DESCRIPTION
**Pull request by jose-bigio**
Signed-off-by: Jose Bigio <jose.bigio@docker.com>

### CODE REVIEW

File: deb/Makefile [hunk 1]
```diff
@@ -157,21 +157,7 @@ sources/dockerd.json: ../common/dockerd.json
...
-
 # offline bundle
 sources/engine.tar: $(DOCKER2OCI)
-	$(MAKE) -C ../image ENGINE_IMAGE=$(ENGINE_IMAGE) ENGINE_SCOPE=$(ENGINE_SCOPE) image-linux
-	mkdir -p artifacts
-	docker save -o artifacts/docker-engine.tar $$(cat ../image/image-linux)
-	./$(DOCKER2OCI) -i artifacts/docker-engine.tar artifacts/engine-image
-	mkdir -p $(@D)
-	tar c -C artifacts/engine-image . > $@
+	$(MAKE) -C ../image ENGINE_IMAGE=$(ENGINE_IMAGE) ENGINE_SCOPE=$(ENGINE_SCOPE) engine-$(ARCH)
```
Comment (corbin-coleman) [inline deb/Makefile]: The target in `image/Makefile` is `engine-$(ARCH).tar`

File: rpm/Makefile [hunk 2]
```diff
@@ -121,21 +121,7 @@ rpmbuild/SOURCES/dockerd.json: ../common/dockerd.json
...
-
 # offline bundle
 rpmbuild/SOURCES/engine.tar: $(DOCKER2OCI)
-	$(MAKE) -C ../image ENGINE_IMAGE=$(ENGINE_IMAGE) ENGINE_SCOPE=$(ENGINE_SCOPE) image-linux
-	mkdir -p artifacts
-	docker save -o artifacts/docker-engine.tar $$(cat ../image/image-linux)
-	./$(DOCKER2OCI) -i artifacts/docker-engine.tar artifacts/engine-image
-	mkdir -p $(@D)
-	tar c -C artifacts/engine-image . > $@
+	$(MAKE) -C ../image ENGINE_IMAGE=$(ENGINE_IMAGE) ENGINE_SCOPE=$(ENGINE_SCOPE) engine-$(ARCH)
```
Comment (corbin-coleman) [inline rpm/Makefile]: Same as comment on `deb/Makeifle`

File: deb/Makefile [hunk 3]
```diff
@@ -157,21 +157,7 @@ sources/dockerd.json: ../common/dockerd.json
...
 # offline bundle
 sources/engine.tar: $(DOCKER2OCI)
-	$(MAKE) -C ../image ENGINE_IMAGE=$(ENGINE_IMAGE) ENGINE_SCOPE=$(ENGINE_SCOPE) image-linux
-	mkdir -p artifacts
-	docker save -o artifacts/docker-engine.tar $$(cat ../image/image-linux)
-	./$(DOCKER2OCI) -i artifacts/docker-engine.tar artifacts/engine-image
-	mkdir -p $(@D)
-	tar c -C artifacts/engine-image . > $@
+	$(MAKE) -C ../image ENGINE_IMAGE=$(ENGINE_IMAGE) ENGINE_SCOPE=$(ENGINE_SCOPE) engine-$(ARCH).tar
+	tar c -C ../engine-$(ARCH) . > $@
```
Comment (corbin-coleman) [inline deb/Makefile]: I missed this before, but the `engine-$(ARCH).tar` target tars the file already.

File: rpm/Makefile [hunk 4]
```diff
@@ -121,21 +121,7 @@ rpmbuild/SOURCES/dockerd.json: ../common/dockerd.json
...
 # offline bundle
 rpmbuild/SOURCES/engine.tar: $(DOCKER2OCI)
-	$(MAKE) -C ../image ENGINE_IMAGE=$(ENGINE_IMAGE) ENGINE_SCOPE=$(ENGINE_SCOPE) image-linux
-	mkdir -p artifacts
-	docker save -o artifacts/docker-engine.tar $$(cat ../image/image-linux)
-	./$(DOCKER2OCI) -i artifacts/docker-engine.tar artifacts/engine-image
-	mkdir -p $(@D)
-	tar c -C artifacts/engine-image . > $@
+	$(MAKE) -C ../image ENGINE_IMAGE=$(ENGINE_IMAGE) ENGINE_SCOPE=$(ENGINE_SCOPE) engine-$(ARCH).tar
+	tar c -C ../image/engine-$(ARCH) . > $@
```
Comment (corbin-coleman) [inline rpm/Makefile]: Ditto other comment

**Comment by seemethere** [comment]
Can this be made to the `master` branch first and then cherry picked in?

**Comment by jose-bigio** [comment]
@seemethere  yeah that's fine. 
Will leave this open until I bring it to the master branch, then I'll close this issue, and cherry-pick the one from master.

**Comment by jose-bigio** [comment]
Closing will Cherry-Pick the changes from `master` https://github.com/docker/docker-ce-packaging/pull/196  

into `18.09` when the `master` changes are merged.

**Pull request closed without merging**